### PR TITLE
🐛  #842: Drop-Downs Allow Deselection of Entries

### PR DESCRIPTION
**Pull request by shailikadakia-oicr**
## Description of Changes
Currently, once a selection has been made in the dropdown, functionality isn't supported to de-select that option. Added the functionality to de-select a select by adding a X (cancel) button to the chosen option. Added some logic to ensure that the errors for de-selecting an option were only applicable to required fields. This way, if a user de-selects an option from on optional field, no error messages are shown.

#### `SelectInput` Component 
- Added a prop `isClearable` which lets users de-select previously selected items in a dropdown. 
- Added a prop `blurInputOnSelect` so the menu us unfocused and closed. This way, users can de-select an option given the X button on the right. 
**Note**: Patrick originally wanted for users to be able to de-select an option while the dropdown menu was focused however that raised some issues with how form validation is currently occurring. Spoke with Patrick and he confirmed the current functionality is good to proceed.

https://github.com/user-attachments/assets/b7619580-187d-455b-b0e3-38c38f03baea

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] [Axe DevTools](https://www.deque.com/axe/browser-extensions/) "Full Page Scan" of updated pages passes without introducing new Critical or Serious issues (UI changes only)
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

**Comment by shailikadakia-oicr** [comment]
Demo at F2F to confirm if we should proceed or debug as to why options can't be unselected when the dropdown menu is focused.

File: apps/consent-ui/src/components/common/Form/fieldsets/SelectFieldSet/index.tsx [hunk 1]
```diff
@@ -55,11 +55,12 @@ const SelectFieldSet = <T extends FieldValues, V extends string>({
 	const idPrefix = useId();
 	const fieldId = `${idPrefix}-${name}`;
 	const descriptionId = `${idPrefix}-description`;
+	const showErrorIfRequired = required ? error : '';
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/SelectFieldSet/index.tsx:58]: are there errors other than the required one? 👀

File: apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx [hunk 2]
```diff
@@ -87,7 +87,8 @@ const SelectInput = <T extends FieldValues, V extends string>({
 							return onChange(onChangeParam);
 						}}
 						onFocus={onFocus}
-						openMenuOnFocus
+						closeMenuOnSelect
```
Comment (samrichca) [inline apps/consent-ui/src/components/common/Form/fieldsets/inputs/SelectInput.tsx:90]: nice, these work great

**Review by samrichca — APPROVED** [review]
working for me, nice work! 🎉 :shipit:

**Pull request merged**